src/Inputs/SearchBar.tsx

```diff
@@ -0,0 +1,236 @@
+import React, {
+    useState,
+    useCallback,
+    useMemo,
+    Children,
+    useLayoutEffect,
+    useRef,
+    RefObject,
+} from 'react';
+import {
+    flex,
+    scroll,
+    position,
+    transition,
+    clickable,
+    darken,
+} from '@Utils/Mixins';
+
+import { LabelLayout, InputFragment as I, LabelLayoutProps } from '@Layouts';
+import { useTransition } from '@Utils/Hooks';
+
+import { Search } from 'styled-icons/fa-solid';
+import styled from 'styled-components';
+
+export interface SearchBarProps extends LabelLayoutProps {
+    suggestiveOptions?: string[];
+    value?: string | number;
+    onChange?: Function;
+    limit?: number;
+    numVisible: number;
+}
+
+export const SearchBar: React.FC<SearchBarProps> = ({
+    suggestiveOptions,
+    value,
+    limit = 4,
+    onChange = (): void => {},
+    name,
+    ...props
+}): React.ReactElement => {
+    const [expanded, setExpanded] = useState(false);
+    const [inputValue, setinputValue] = useState('');
+    const [, mount, animation] = useTransition(expanded);
+    const refSelectList = useRef() as RefObject<HTMLDivElement>;
+    const [numVisibleSelection, setNumVisibleSelection] = useState(limit);
```

Comment:
This is very odd, why are you setting the state to the prop value??

Reply:
numVisibleSelection is not a prop anymore, I changed it to state, I think limit would do the job and I used numVisibleSelection to get the number of children inside the Selection List and the minimum of limit or  numVisibleSelection gets effect. So that when there are for example only 2 results of the suggestions the SelectList doesn't expand more than that

Reply:
But why is this a state? Why not just calculate this in either a ref if its never changing or inside the component that is changing every click.

Reply:
Well it gets the length from the visible selectitems inside the SelectList which changes with every keydown, , so it needs the SelectList to be mounted so that it can get the current number of children inside of it. thats why I put the setNumVisibleSelection inside the useLayoutEffect

Reply:
Ok that makes sense. Let’s merge this then.